src/main/java/com/synopsys/integration/alert/web/api/config/ConfigActions.java

```diff
@@ -118,54 +160,103 @@ public void deleteConfig(Long id) throws AlertException {
-        Collection<ConfigurationFieldModel> updatedFields = fieldModelProcessor.fillFieldModelWithExistingData(id, updatedFieldModel);
-        ConfigurationModel configurationModel = configurationAccessor.updateConfiguration(id, updatedFields);
-        FieldModel dbSavedModel = modelConverter.convertToFieldModel(configurationModel);
-        FieldModel afterUpdateAction = fieldModelProcessor.performAfterUpdateAction(previousFieldModel, dbSavedModel);
-        return dbSavedModel.fill(afterUpdateAction);
+    @Override
+    protected ActionResponse<FieldModel> updateAfterChecks(Long id, FieldModel resource) {
+        try {
+            Optional<ConfigurationModel> optionalPreviousConfig = configurationAccessor.getConfigurationById(id);
+            FieldModel previousFieldModel = optionalPreviousConfig.isPresent() ? modelConverter.convertToFieldModel(optionalPreviousConfig.get()) : null;
+
+            FieldModel updatedFieldModel = fieldModelProcessor.performBeforeUpdateAction(resource);
+            Collection<ConfigurationFieldModel> updatedFields = fieldModelProcessor.fillFieldModelWithExistingData(id, updatedFieldModel);
+            ConfigurationModel configurationModel = configurationAccessor.updateConfiguration(id, updatedFields);
+            FieldModel dbSavedModel = modelConverter.convertToFieldModel(configurationModel);
+            FieldModel afterUpdateAction = fieldModelProcessor.performAfterUpdateAction(previousFieldModel, dbSavedModel);
+            FieldModel responseModel = dbSavedModel.fill(afterUpdateAction);
+            return new ActionResponse<>(HttpStatus.OK, responseModel);
+        } catch (AlertException ex) {
+            logger.error("Error creating configuration", ex);
+            return new ActionResponse<>(HttpStatus.INTERNAL_SERVER_ERROR, String.format("Error creating config: %s", ex.getMessage()));
+        }
     }
 
-    public String validateConfig(FieldModel fieldModel, List<AlertFieldStatus> fieldErrors) throws AlertFieldException {
-        fieldErrors.addAll(fieldModelProcessor.validateFieldModel(fieldModel));
-        if (!fieldErrors.isEmpty()) {
-            throw new AlertFieldException(fieldErrors);
+    @Override
+    protected ValidationActionResponse validateAfterChecks(FieldModel resource) {
+        List<AlertFieldStatus> fieldStatuses = fieldModelProcessor.validateFieldModel(resource);
+        ValidationResponseModel responseModel;
+        HttpStatus status = HttpStatus.OK;
+        if (fieldStatuses.isEmpty()) {
+            responseModel = ValidationResponseModel.withoutFieldStatuses("The configuration is valid");
+        } else {
+            status = HttpStatus.BAD_REQUEST;
+            responseModel = ValidationResponseModel.fromStatusCollection("There were problems with the configuration", fieldStatuses);
         }
-        return "Valid";
+        return new ValidationActionResponse(status, responseModel);
```

Comment:
This may look better if we fo an if check with a return then return Bad request by default if the if statement is skipped.

Reply:
There's a ticket to add audit logging especially for users and role actions.  To ensure that we are logging consistently we most likely won't be able to use the circuit breaker pattern and return right away if we want to log when an action starts and completes.

Reply:
Won't we still be able to log in all cases? Returning early would act like the if else statement above and you can fit all logs as needed, unless I'm missing something about that logging.